tests/entrypoint/database/test_init.py

```diff
@@ -25,3 +27,18 @@ def test_sqlite(tmpdir):
     main(args_and_opts)
 
     assert db_path.exists()
+
+    engine = sqlalchemy.create_engine(f"sqlite:///{db_path}")
+    metadata = sqlalchemy.MetaData(engine)
+    metadata.reflect(engine)
+    tables = metadata.sorted_tables
+
+    expected_metadata = sqlalchemy.MetaData()
+    schema_articles(expected_metadata)
+    schema_sentences(expected_metadata)
+    expected_tables = expected_metadata.sorted_tables
+
+    assert len(tables) == len(expected_tables)
+
+    for table, expected in zip(tables, expected_tables):
+        assert table.compare(expected)
```

Comment:
I find this test to be a tautology since it is always going to pass. It feels very similar to:

```
def foo():
     ...

def test_foo():
    assert foo() == foo()
```
IMO we are abusing the fact that we know what is happening inside of `bbs_databse init` (we use the helper `schema_articles` and `schema_sentences`).

Don't you think it would be more useful to just hardcode the column names inside of the tests and then check for set equality?

Reply:
> I find this test to be a tautology since it is always going to pass.

Even with line 33, `metadata.reflect(engine)`?

Indeed, `reflect(...)` loads all available table definitions from the database ([SQLAlchemy docs](https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.MetaData.reflect)).

So, the test really checks that the database has the expected tables and table schemas. Therefore, it doesn't seem that the test does `assert foo() == foo()`.

Or where you thinking about something else?

Reply:
But how can this test ever fail? We are literally importing and using `schema_articles` and `schema_sentecnes` in the test which is exactly the same thing we are doing in the source code?

Reply:
> But how can this test ever fail?

If `bbs_database init` does not initialize the database? For example, `metadata.create_all` is not called.

The two function `schema_articles` and `schema_sentences` only define the table schemas. They do not initialize the database.

If with this clarifications you still think that's better to check only for the column names and to hardcode them, then it will be done. That will be less maintainable though.

Reply:
Let's see what @Stannislav @FrancescoCasalegno think:) I am completely willing to make a compromise:) I definitely do see your point of viewing the `schema.py` as a config file rather than source code.

I just think that it would nice to have a failing test if somebody tries to change the schema. Additionally, I think that the asserts inside of unit tests should be as explicit as possible (=enumerating the expected columns) rather

Reply:
Sooo, my 5 cents to put more fuel to the fire 😆.

Maybe it's better to ask: what kind of behaviour are we trying to test here?

If the test hypothesis is "Test that `run` creates a database under given path and populates it with schemas that reflect `schema_articles` and `schema_sentences`" then maybe it's fine.

If, on the other hand, we want to guarantee a certain schema structure then maybe we can have separate tests for `schema_articles` and `schema_sentences`.

Reply:
Discussed during standup:) Let's keep it the way it is and potentially write a test that really tests the schema at one point in the future.